### DISCUSSION TITLE
iOS: Perform Action on clicking Notification

### DISCUSSION THREAD

**Discussion by Excel1** [capawesome-team/capacitor-firebase#742] [Q&A] [answered]
My device receive a notification with a deeplink in the data object. Now i want, that when a user click the notification, the deeplink will performed and opens the correct page in the capacitor app.

On Android you can build your notification and add the link to it. Everything works as expected. But on iOS i dont know how to perform the action by on click (notification). 

I already send the link and convert it in the NotificationServiceExtension to a deeplink. I tried to catch the didReceive Event but without any results. How can i implement a simple method in my app delegate to open a deeplink send by firebase?

**Comment by robingenz** [comment, marked as answer]
I always send the link as a data payload with the notification. Then I listen for the `notificationActionPerformed` event and read the link from the data payload and navigate. No need to convert anything in the NotificationServiceExtension. At least that's my approach.

**Reply by Excel1** [reply]
@robingenz Thanks for sharing your experience. I need the ServiceExtension to translate my notification and set the link up there. I will try tomorrow to test it without editing link in the extension and listening on notification action performed event. Do you listen directly in the App Delegate?

**Reply by robingenz** [reply]
I don't use any custom native code for this except the plugin itself. Only the `notificationActionPerformed` listener.

**Reply by Excel1** [reply]
@robingenz It works perfectly - I had probably used it once but removed it because it is easy to implement natively on android but it works even better this way. Many thanks for the tip!
